Docs/themes

xylish7 opened the pull request:
Closes #3066

## 📝 Description

This PR introduces a theme generator to the NextUI documentation. The theme generator is accessible via a new navigation item, `Themes`, making it easier for users to create and customize themes.

![image](https://github.com/user-attachments/assets/5ba737a5-83eb-4449-a8ea-a71c0db541f7)

## ⛳️ Current Behavior

Currently, the NextUI documentation does not include a theme generator, limiting users' ability to easily create and preview custom themes.

## 🚀 New Behavior

With this PR, a theme generator will be added to the NextUI documentation, allowing users to generate and customize themes directly within the platform.

## 💣 Breaking Change

No, this PR does not introduce any breaking changes.

## 📝 Additional Information

There is an issue with the default color in the white theme; it does not perfectly match the tints and shades of the zinc color used by NextUI's default palette. Although a suitable solution hasn't been found yet, this should not be a blocker, as users can manually adjust the colors. Efforts to resolve this issue will continue.

This version is more concise and polished, improving clarity and flow while retaining all important details.

wingkwong reviewed (changes requested):
1. please resolve coderabbitai comments. 
2. [typecheck](https://github.com/nextui-org/nextui/actions/runs/11081293780/job/30792890832?pr=3707) is failed. please take a look

xylish7 commented:
@wingkwong I fixed the ts error and also implemented the feedback from coderabbitai, but the new reviews from coderabbit ai feels a little bit overkill and don't think they improve the quality of code. I don't find that level of abstraction needed for this simple case.

wingkwong commented:
Thanks. We'll review gradually.

wingkwong reviewed (changes requested):
docs s failed to build. please double check.

xylish7 commented:
Hmm...it's weird why it's complaining about brand color not being defined since...since the types seems to be correct. Will look into it.

xylish7 commented:
I'm unable to see why there is a issue with that. 

The `Config` interface looks like this
```jsx
export interface Config {
  light: ConfigColors;
  dark: ConfigColors;
  layout: ConfigLayout;
}
```
The `ThemeType` type loooks like this:
```jsx
export type ThemeType = "light" | "dark";
```
And the issue is mentioned in [brand-colors.tsx](https://github.com/xylish7/nextui/blob/cb1a2f372aba2609f408dc9c7d96b3daef603777/apps/docs/components/themes/components/configuration/brand-colors.tsx#L22), where it says that `config[theme]` is undefined, which, based on the types definition of `Config` and `ThemeType` it can't be true. 

![image](https://github.com/user-attachments/assets/b4475940-33b2-4f5c-96de-60b77ac988ca)

Can it be a Typescript compiler issue or I'm missing something?

xylish7 commented:
So the problem was related to SSR. The theme builder was rendered on server side and because of this some values were initialized with false. I've added a check to render the theme builder only on client side and while the js is loading, a loader is displayed.

![image](https://github.com/user-attachments/assets/7cc687a6-bd4c-4d53-8b26-d3fd736d46e6)

jrgarciadev commented:
Hey @xylish7 thanks for this great contribution 🙏🏻 could you please fix the conflicts?

xylish7 commented:
Sure! Will look into it today 👍🏻

xylish7 commented:
@jrgarciadev I've fixed the merge conflicts. Only `pnpm-lock.yaml` was conflicting.

I also run the `build` and `test` command which successfully ran, but there was an error related to the usage of `new Image` which I haven't touched, so not sure if that was already present or it was because I regenerated the `pnpm-lock.yaml` file

macci001 commented:
handled in #4626

xylish7 commented:
Thanks for your further contributions and improvements. Can't wait to see it in prod!
